core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -1357,7 +1357,7 @@ private[spark] object JsonProtocol {
           message.getOrElse("Unknown reason"))
       case `exceptionFailure` =>
         val className = json.get("Class Name").extractString
-        val description = json.get("Description").extractString
+        val description = jsonOption(json.get("Description")).map(_.extractString).orNull
```

Comment:
When can this be null ?
(This was the description i was referring to in my comment earlier - should have made it here, instead of general comment :-( )

Reply:
I forget the context of this change. Let me revert this change, and double confirm. if this is needed, I'll send another PR.